edb/lang/edgeql/ast.py

```diff
@@ -232,29 +232,74 @@ class FunctionCall(Expr):
     agg_set_modifier: typing.Optional[SetModifier]
 
 
-class Constant(Expr):
-    value: typing.Union[int, str, float, bool, bytes, decimal.Decimal]
+class BaseConstant(Expr):
+    value: str
+
+    @classmethod
+    def from_python(cls, val: object) -> 'BaseConstant':
+        if isinstance(val, str):
+            return StringConstant.from_python(val)
+        elif isinstance(val, bool):
+            return BooleanConstant(value='true' if val else 'false')
+        elif isinstance(val, int):
+            return IntegerConstant(value=str(val))
+        elif isinstance(val, decimal.Decimal):
+            if val.to_integral_value() == val:
+                return IntegerConstant(value=str(val))
+            else:
+                return FloatConstant(value=str(val))
+        elif isinstance(val, float):
+            return FloatConstant(value=str(val))
```

Comment:
Should we use `val.hex()` for floats?

Reply:
IIRC, `from_python(<float>)` is only used in the GraphQL translator. I'll remove support for float coercion altogether once the GraphQL AST gets the same treatment as here.